packages/data-provider/test/utils/convertSqonToQuery.spec.ts

```diff
@@ -0,0 +1,243 @@
+import { expect } from 'chai';
+import { SQL, SQLChunk } from 'drizzle-orm';
+import { describe, it } from 'mocha';
+
+import { SQON } from '@overture-stack/sqon-builder';
+
+import { convertSqonToQuery, parseSQON } from '../../src/utils/convertSqonToQuery.js';
+
+function extractValues(obj: SQL | undefined, key: string) {
+	let values: SQL[] = [];
+
+	function recurse(currentObj: SQL | undefined) {
```

Comment:
this function needn't be defined withing extractValues, as far as I can tell.

Reply:
Added types on this function as mocha tries to do typescript checks and couldn't find a way to exclude  on test files.

The purpose of this function is to convert the input SQL object from drizzle into an Array which make it easier to compare on `expect()` function

Reply:
to be clear, what I mean is lines 18-28 could be cut and pasted to line 8 (outside `extractValues`) and the result would be the same, minus the reinstantiation of `recurse` every time the parent function runs